lib/Dialect/ModArith/IR/ModArithOps.h

```diff
@@ -3,11 +3,44 @@
 
 // NOLINTBEGIN(misc-include-cleaner): Required to define ModArithOps
 #include "lib/Dialect/ModArith/IR/ModArithDialect.h"
+#include "lib/Dialect/ModArith/IR/ModArithTypes.h"
 #include "mlir/include/mlir/IR/BuiltinOps.h"  // from @llvm-project
 #include "mlir/include/mlir/Interfaces/InferTypeOpInterface.h"  // from @llvm-project
 // NOLINTEND(misc-include-cleaner)
 
 #define GET_OP_CLASSES
 #include "lib/Dialect/ModArith/IR/ModArithOps.h.inc"
 
+namespace mlir {
+namespace heir {
+namespace mod_arith {
+
+template <typename OpType>
+inline ModArithType getResultModArithType(OpType op) {
+  return llvm::cast<ModArithType>(
```

Comment:
Nit: I believe you can omit the `llvm::` prefix because the mlir namespace re-imports them under `mlir::`

```suggestion
  return cast<ModArithType>(
```